cupy/random/_generator_api.pyx

```diff
@@ -237,8 +237,36 @@ class Generator:
             :meth:`numpy.random.Generator.beta`
         """
         cdef ndarray y
-        y = ndarray(size if size is not None else (), numpy.float64)
-        _launch_dist(self.bit_generator, beta, y, (a, b))
+        cdef a_arr, b_arr
+
+        if not isinstance(a, ndarray):
+            if type(a) in (float, int):
+                a = cupy.asarray(a, numpy.float64)
+            else:
+                raise TypeError('a is required to be a cupy.ndarray'
+                                ' or a scalar')
+        if not isinstance(b, ndarray):
+            if type(b) in (float, int):
+                b = cupy.asarray(b, numpy.float64)
+            else:
+                raise TypeError('b is required to be a cupy.ndarray'
+                                ' or a scalar')
+
+        if size is not None and not isinstance(size, tuple):
+            size = (size, )
+        elif size is None:
+            size = cupy.broadcast(a, b).shape
+
+        y = ndarray(size, numpy.float64)
+
+        a = cupy.broadcast_to(a, y.shape)
+        b = cupy.broadcast_to(b, y.shape)
+        a_arr = _array_data(a)
+        b_arr = _array_data(b)
```

Comment:
What do the ndarrays returned by `_array_data` contain?

Reply:
Here a_arr will have value `cupy.array((a.data.ptr, a.ndim) + a.shape + a.strides)`

Reply:
`a` and `b` are still of `double` in `beta()`'s arguments. You should fix their types to `intptr_t` to make it take `a_arr.data.ptr` and `b_arr.data.ptr`.

Also, You need to cast them to `int64_t*` to pass to `generator_dispatcher` in `beta()`, otherwise the second form of `get_index` (with `std::is_arithmetic<T>`) is selected in `execute_dist`, eventually the pointer values are casted to `double` and used for `a` and `b` to compute `rk_beta`.